File: solutions/0003/substring_no_repeat.c
int lengthOfLongestSubstring(char * s){
    
    int seen[128] = {0};
    char *first = s;
    char *curr = s;
    
    while (*curr){
        char my_char = *curr++;
        seen[my_char]++;
        bool unique = true;
        
        for (char *c = first; c <= curr; c++){
            if (seen[*c] > 1){
                unique = false;
                break;
            }
        }
        
        if (!unique){
            seen[*first++]--;
        }
    }
    
    return curr - first;
}

File: solutions/0003/substring_no_repeat.py
from collections import Counter

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        start = 0
        size = len(s)
        seen = Counter()
        
        for char in s:
            seen.update(char)
            top_char = seen.most_common(1)[0][1]
            if top_char > 1:
                seen.subtract(s[start])
                start += 1
                
        return size - start
